third_party/terraform/data_sources/data_source_google_iam_testable_permissions.go

```diff
@@ -0,0 +1,136 @@
+package google
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+)
+
+func dataSourceGoogleIamTestablePermissions() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceGoogleIamTestablePermissionsRead,
+		Schema: map[string]*schema.Schema{
+			"full_resource_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"stages": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validation.StringInSlice([]string{"ALPHA", "BETA", "GA", "DEPRECATED"}, true),
+				},
+			},
+			"custom_support_level": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "SUPPORTED",
+				ValidateFunc: validation.StringInSlice([]string{"NOT_SUPPORTED", "SUPPORTED", "TESTING"}, true),
+			},
+			"permissions": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"title": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"custom_support_level": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"stage": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"api_disabled": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceGoogleIamTestablePermissionsRead(d *schema.ResourceData, meta interface{}) (err error) {
+	config := meta.(*Config)
+	body := make(map[string]interface{})
+	body["pageSize"] = 500
+	permissions := make([]map[string]interface{}, 0)
+
+	custom_support_level := strings.ToUpper(d.Get("custom_support_level").(string))
+	stages := []string{}
+	for _, e := range d.Get("stages").([]interface{}) {
+		stages = append(stages, strings.ToUpper(e.(string)))
```

Comment:
You shouldn't need ToUpper here as the validate func should enforce that it is uppercase

Reply:
I'd like to make it case-insensitive so users can specify `beta` or `ga` without errors, hence the `true` in the validation, but then I need to use `ToUpper` to ensure it matches the API response.